pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -431,7 +390,15 @@ func registerHandlers(informer cache.SharedIndexInformer, q queue.Instance, otyp
 				})
 			},
 			UpdateFunc: func(old, cur interface{}) {
-				if !equalsFunc(old, cur) {
+				oldObj := old.(metav1.Object)
+				newObj := cur.(metav1.Object)
+				if oldObj.GetResourceVersion() == newObj.GetResourceVersion() {
```

Comment:
Mote: instead of calling reflect.DeepEqual, this is more efficient

Reply:
Can we define a VersionBasedEqualsFunc outside and let it be passed here - instead of doing a special handling of nil and this here?

Reply:
Let me think

Reply:
Now I do not check nil here, set the func default to resource version check if not provided